nvflare/apis/utils/reliable_message.py

```diff
@@ -337,6 +342,7 @@ def release_request_receiver(cls, receiver: _RequestReceiver, fl_ctx: FLContext)
         """
         with cls._tx_lock:
             cls._req_receivers.pop(receiver.tx_id, None)
+            cls._register_completed_req(receiver.tx_id, receiver.tx_timeout)
```

Comment:
We should do this before popping it from cls._req_receivers, in case another request is received right after the pop and before added to completed_reqs.

Reply:
Good catch. Reversed.